openapi/controller/main.py

```diff
@@ -0,0 +1,183 @@
+# -*- coding: utf-8 -*-
+import functools
+import hashlib
+import logging
+import os
+from ast import literal_eval
+import base64
+
+import odoo
+from odoo import http, SUPERUSER_ID, models
+from odoo.http import request, OpenERPSession
+from odoo.modules.registry import RegistryManager
+from psycopg2.extensions import ISOLATION_LEVEL_READ_COMMITTED
+
+from .pinguin import *
+
+_logger = logging.getLogger(__name__)
+
+#################################################################
+# Odoo REST API                                                 #
+#  Version 1                                                    #
+# --------------------------------------------------------------#
+# The current api version is considered stable, although        #
+# the exposed models and methods change as they are configured  #
+# on the database level. Only if significant changes in the api #
+# generation logic should be implemented in the future          #
+# a version bump should be considered.                          #
+#################################################################
+
+API_ENDPOINT = '/api'
+API_ENDPOINT_V1 = '/v1'
+# API_ENDPOINT_V2 = '/2'
+
+# We patch the route decorator in pinguin.py
+# with authentication and DB inference logic.
+# We also check if the model is installed in the database.
```

Comment:
>    We also check if the **module** is installed in the database.

?

I'd prefer just to use ``post_load`` to be sure that http really has to be patched

Reply:
pateched `route` should contain all the permission and resource level checks before going on...
This file should be just a coded interface, no actual logic, so I wanted to package it away...